R/stat-smooth.R

```diff
@@ -171,14 +171,25 @@ StatSmooth <- ggproto("StatSmooth", Stat,
       method.args$method <- "REML"
     }
 
-    model <- inject(method(
-      formula,
-      data = data,
-      weights = weight,
-      !!!method.args
-    ))
-
-    prediction <- predictdf(model, xseq, se, level)
+    prediction <- tryCatch(
+      {
+        model <- inject(method(
+          formula,
+          data = data,
+          weights = weight,
+          !!!method.args
+        ))
+        predictdf(model, xseq, se, level)
+      },
+      error = function(cnd) {
+        cli::cli_warn("Failed to fit group {data$group[1]}.", parent = cnd)
+        NULL
+      }
+    )
+    if (is.null(prediction)) {
+      return(NULL)
+    }
+
```

Comment:
We prefer to use the slang error handling when possible (`try_fetch()`)

Reply:
Thanks for pointing this out. I'm not up to speed with what the difference exactly is, but that seems to work well here too.